rdkcentral/Dobby
# Run plugin hooks in order determined by dependencies

### Description
DobbyRdkPluginManager now processes RDK plugins for a given hook according to the order of plugin dependency. 
Before this change, RDK plugins were processed according to the order in which they are declared in the "RdkPlugins" section of the "defs-plugins" schema file. This can result in hook execution failures when plugin A which depends on plugin B is processed before plugin B. 

After the change, if plugin A depends on plugin B, hooks for plugin B will be run before hooks for plugin A.
This is inverted (A before B) for postHalt and postStop so that the plugins on which other plugins depend are shut down later.

The dependencies can be defined in the container spec. Each plugin spec now contains a "dependsOn" array.
If, for example, plugin A depends on plugins B and C, then plugin A's spec can look like this:
"a": {
    "required": false,
    "dependsOn": [ "b", "c" ],
    "data": {}
}

Dependency cycles are detected and make DobbyRdkPluginManager abort before it starts running the plugin hooks.

### Test Procedure
Create a container spec with "dependsOn" clauses defined for some plugins, use DobbyPluginLauncher to check if hooks are executed according to the order of dependency.

### Type of Change

### Requires Bitbake Recipe changes?
